packages/core/src/asciidoctor-core-api.js

```diff
@@ -3047,3 +3047,402 @@ Opal.Asciidoctor.SyntaxHighlighterBase = SyntaxHighlighterBase
+  let ret = this.getAttributes()[name]
+  return (ret !== Opal.nil) ? ret : undefined
+}
+
+/**
+ * Get the caption of the table
+ * @returns {string}
+ */
+Table.prototype.getCaption = function () {
+  return this.caption
+}
+
+/**
+ * Get the rows of this table
+ * @returns {Object} An object with the members "head", "body" and "foot"
+ */
+Table.prototype.getRows = function () {
+  return this.rows
+}
+
+/**
+ * Get the columns of this table
+ * @returns {Array<Column>}
+ */
+Table.prototype.getColumns = function () {
+  return this.columns
+}
+
+/**
+ * Get the head rows of this table
+ * @returns an Array of Array of Cell
+ */
+Table.prototype.getHead = function () {
+  return this.rows.head
+}
+
+/**
+ * Get the body rows of this table
+ * @returns an Array of Array of Cell
+ */
+Table.prototype.getBody = function () {
+  return this.rows.body
+}
+
+/**
+ * Get the foot rows of this table
+ * @returns an Array of Array of Cell
+ */
+Table.prototype.getFoot = function () {
+  return this.rows.foot
+}
+
+/**
+ * Check if the table has rows
+ * @returns {boolean}
+ */
+Table.prototype.hasRows = function () {
+  return (this.rows !== Opal.nil) && (this.rows.body.length > 0)
+}
+
+/**
+ * Check if the table has a header
+ * @returns {boolean}
+ */
+Table.prototype.hasHeader = function () {
+  return (this.rows !== Opal.nil) && (this.rows.head.length > 0)
+}
+
+/**
+ * Check if the table has a header option set
+ * @returns {boolean}
+ */
+Table.prototype.hasHeaderOption = function () {
+  return this.has_header_option
+}
+
+/**
+ * Check if the table has a footer
+ * @returns {boolean}
+ */
+Table.prototype.hasFooter = function () {
+  return (this.rows !== Opal.nil) && (this.rows.foot.length > 0)
+}
+
+/**
+ * Check if the table has the footer option set
+ * @returns {boolean}
+ */
+Table.prototype.hasFooterOption = function () {
+  return this.hasOption('footer')
+}
+
+/**
+ * Check if the table has the autowidth option set
+ * @returns {boolean}
+ */
+Table.prototype.hasAutowidthOption = function () {
+  return this.hasOption('autowidth')
+}
+
+/**
+ * Get the number of rows in the table. Please be carefull, the
+ * header and footer rows are also counted
+ * @returns {string|undefined}
+ */
+Table.prototype.getRowcount = function () {
+  return this.getAttribute('rowcount')
+}
+
+/**
+ * Get the number of columns in the table
+ * @returns {string|undefined}
+ */
+Table.prototype.getColcount = function () {
+  return this.getAttribute('colcount')
+}
+
+// Table Column
+
+/**
+ * Methods to manage the columns of an AsciiDoc table. In particular, it
+ * keeps track of the column specs
+ * @namespace
+ * @extends AbstractNode
+ */
+var Column = Opal.Asciidoctor.Table.Column
+
+/**
+ * Create a new Column element
+ * @param {Table} table
+ * @param {number} index
+ * @param {Object|undefined} attributes
+ * @returns Column
+ */
+Column.create = function (table, index, attributes) {
+  let ret = this.$new(table, index, attributes)
+  return ret
```

Comment:
It could be inlined right?

```suggestion
  return this.$new(table, index, attributes)
```

Reply:
Done. Will be included in the next commit